scripts/Inelastic/IndirectReductionCommon.py

```diff
@@ -755,7 +763,7 @@ def group_spectra_of(workspace, masked_detectors, method, group_file=None, group
         if len(masked_detectors) > 0:
             _mask_detectors(workspace, masked_detectors)
 
-        group_string = create_detector_grouping_string(number_of_groups, 0, workspace.getNumberHistograms() - 1)
+        group_string = create_detector_grouping_string(number_of_groups, spectra_range[0], spectra_range[1])
```

Comment:
Can you maybe add another assert on this function to prevent that NGroups>spectra_max-spectra_min?

Reply:
Good point, I've added an assertion and a validation against this happening at the interface level